### PR TITLE
New publish command

### LINKED ISSUE #255: Move --full flag to a new publish command
After running `publish` command, all the proxies created should be transferred to the `App`. This will allow the user to keep using the CLI as expected, otherwise running `update` for any of those proxies will fail

### PR DESCRIPTION
**Pull request by spalladino**
Adds a new `publish` command that migrates the current app/lib from lightweight into a full app, by deploying a directory, package and app contracts.

Also deprecates the `--full` flag for `push`, and renames that flag in `init` to `--publish`.

Fixes #255

### CODE REVIEW

**Comment by spalladino** [comment]
Pending transfering ownership of proxies to app. Can be merged in the meantime though.

**Comment by spalladino** [comment]
@facuspagnuolo added proxies transfer, ready to review now. I'll rebase onto master in a few minutes.

File: packages/cli/src/commands/init.js [hunk 1]
```diff
@@ -11,18 +11,18 @@ const register = program => program
   .command(signature, { noHelp: true })
   .usage('<project-name> [version]')
   .description(description)
-  .option('--full', 'create an application, package, and implementation directory contracts to manage your project')
+  .option('--publish', 'automatically publishes your application upon pushing it to a network')
```
Comment (facuspagnuolo) [inline packages/cli/src/commands/init.js]: I'd use the same description you used for the `publish` command. Note that you are using `project` for the other and here you are using `application`

File: packages/cli/src/models/files/ZosPackageFile.js [hunk 2]
```diff
@@ -103,8 +99,8 @@ export default class ZosPackageFile {
     this.data.lib = lib
   }
 
-  set full(full) {
-    this.data.full = !!full
+  set publish(publish) {
+    this.data.publish = !!publish
```
Comment (facuspagnuolo) [inline packages/cli/src/models/files/ZosPackageFile.js]: IMO it should say `published` but I'm not sure if we are using that convention

File: packages/lib/src/utils/Transactions.js [hunk 3]
```diff
@@ -184,3 +184,25 @@ async function calculateActualGas(estimatedGas) {
...
+  while (true) {
+    if ((new Date() - now) > timeout) {
+      throw new Error(`Exceeded timeout of ${timeout / 1000} seconds awaiting confirmations for transaction ${transactionHash}`)
+    }
+    const currentBlock = await getCurrentBlock();
+    const txBlock = await getTxBlock();
+    if (currentBlock - txBlock >= confirmations) return true;
+    await sleep(interval);
+  }
+}
```
Comment (facuspagnuolo) [inline packages/lib/src/utils/Transactions.js]: noice!

File: packages/vouching/src/scripts/deploy.js [hunk 4]
```diff
@@ -6,15 +6,16 @@ import configureTPL from '../kernel/configureTPL'
...
 
 export default async function deploy(options) {
   const oneDay = 60 * 60 * 24
   session({ expires: oneDay, ...options })
   log.base(`Pushing ZeppelinOS app with options ${JSON.stringify(options, null, 2)}...`)
   const isLocalOrTest = options.network === 'local' || options.network === 'test'
   if (isLocalOrTest) removeZosFiles(options)
-  await push({ deployLibs: isLocalOrTest, full: true, ...options })
+  await push({ deployLibs: isLocalOrTest, ...options })
+  await publish({ ...options })
```
Comment (facuspagnuolo) [inline packages/vouching/src/scripts/deploy.js]: 🙌

**Pull request merged**